### PR TITLE
Chapter1/homework/7

### PR DESCRIPTION
**Pull request by TelefunkenU87**
Lets see what Codacity and Azure Pipelines thinks about this.

### CODE REVIEW

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 17
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/BalanceStats.cs  4
- Src/BootCamp.Chapter/TextTable.cs  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603&bid=16933732)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 1]
```diff
@@ -1,37 +1,261 @@
...
+        {
+            string[][] workingPeopleAndBalances = ArrayifyBalances(peopleAndBalances);
+            string totalNames = "";
+            for (int i = 0; i < workingPeopleAndBalances.Length; i++)
+            {
+                float[] balances = StringArrToFloatArr(workingPeopleAndBalances[i][1..]);
+                bool isBalance = false;
+                foreach (var item in balances)
+                {
+                    if (item == balance) isBalance = true;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Do not check floating point equality with exact values, use a range instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 2]
```diff
@@ -1,37 +1,261 @@
...
+        /// <summary>
+        /// Takes an array of floats and returns a single float of the highest value.
+        /// </summary>
+        public static float HighestBalance(float[] balances)
+        {
+            float highestBalance = balances[0];
+            if (balances.Length == 0) return highestBalance;
+            for (int i = 1; i < balances.Length; i++)
+            {
+                if (balances[i] > highestBalance) highestBalance = balances[i];
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 3]
```diff
@@ -1,37 +1,261 @@
...
+        /// <summary>
+        /// Takes a string of names separated by ', ' and returns a grammatically correct string
+        /// </summary>
+        /// <param name="balanceNames"></param>
+        /// <returns></returns>
+        public static string FormatBalanceNames(string balanceNames)
+        {
+            int count = balanceNames.Split(',').Length - 1;
+            balanceNames = balanceNames.Remove(balanceNames.Length - 2);
+            if (count == 1) return balanceNames;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 4]
```diff
@@ -1,37 +1,261 @@
...
+        /// Takes an array of strings and returns that array as floats.
+        /// </summary>
+        public static float[] StringArrToFloatArr(string[] balances)
+        {
+            int failLength = 0;
+            float[] floatArr = new float[balances.Length];
+            for (int i = 0; i < floatArr.Length; i++)
+            {
+                bool success = float.TryParse(balances[i], out float number);
+                if (success) floatArr[i - failLength] = number;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 5]
```diff
@@ -1,37 +1,261 @@
...
+                return message;
+            }
+            for (int i = 1; i < workingBalances.Length; i++)
+            {
+                if (float.Parse(workingBalances[i][^1]) < float.Parse(workingBalances[poorestPerson][^1]))
+                {
+                    poorestPerson = i;
+                }
+            }
+            return message = $"{workingBalances[poorestPerson][0]} is the poorest person. ${workingBalances[poorestPerson][^1]}";
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Remove this useless assignment to local variable 'message'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/TextTable.cs [hunk 6]
```diff
@@ -34,7 +36,48 @@ public static class TextTable
...
+            {
+                sb.Append("|").Append(' ', splitMessage[longStringIndex].Length + padding*2).AppendLine("|");
+            }
+            sb.Append("+").Append('-', splitMessage[longStringIndex].Length + padding*2).AppendLine("+");
+
+            return sb.ToString();
+        }
+        public static int LongestStringIndex(string[] message)
+        {
+            if (message.Length == 1) return 0;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/TextTable.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 7]
```diff
@@ -1,37 +1,261 @@
...
+        /// <summary>
+        /// Takes an array of names and balances, then returns the highest value of all balances.
+        /// </summary>
+        /// <param name="peopleAndBalances"></param>
+        /// <returns></returns>
+        public static float HighestBalanceEver(string[] peopleAndBalances)
+        {
+            string[][] workingPeopleAndBalances = ArrayifyBalances(peopleAndBalances);
+            float highestBalanceEver = HighestBalance(StringArrToFloatArr(workingPeopleAndBalances[0][1..]));
+            if (workingPeopleAndBalances.Length == 1) return highestBalanceEver;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 8]
```diff
@@ -1,37 +1,261 @@
...
+            return failFloatArr;
+        }
+
+        /// <summary>
+        /// Takes an array of floats and returns a single float of the highest value.
+        /// </summary>
+        public static float HighestBalance(float[] balances)
+        {
+            float highestBalance = balances[0];
+            if (balances.Length == 0) return highestBalance;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 9]
```diff
@@ -1,37 +1,261 @@
...
+            string totalNames = "";
+            for (int i = 0; i < workingPeopleAndBalances.Length; i++)
+            {
+                float[] balances = StringArrToFloatArr(workingPeopleAndBalances[i][1..]);
+                bool isBalance = false;
+                foreach (var item in balances)
+                {
+                    if (item == balance) isBalance = true;
+                }
+                if (isBalance) totalNames += $"{workingPeopleAndBalances[i][0]}, ";
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/TextTable.cs [hunk 10]
```diff
@@ -34,7 +36,48 @@ public static class TextTable
         /// </summary>
         public static string Build(string message, int padding)
         {
-            return "";
+            if (string.IsNullOrEmpty(message))
+            {
+                return "";
+            }
+
+            string[] splitMessage = message.Split(new string[] { System.Environment.NewLine }, System.StringSplitOptions.None);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/TextTable.cs]: Codacy Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 11]
```diff
@@ -1,37 +1,261 @@
...
+        /// <param name="balanceNames"></param>
+        /// <returns></returns>
+        public static string FormatBalanceNames(string balanceNames)
+        {
+            int count = balanceNames.Split(',').Length - 1;
+            balanceNames = balanceNames.Remove(balanceNames.Length - 2);
+            if (count == 1) return balanceNames;
+            string[] names = balanceNames.Split(',');
+            StringBuilder sbNames = new StringBuilder(names[0]);
+            switch (count)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Replace this 'switch' statement with 'if' statements to increase readability.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs (hunk 9 above)
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Use a StringBuilder instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 12]
```diff
@@ -1,37 +1,261 @@
...
+        {
+            int failLength = 0;
+            float[] floatArr = new float[balances.Length];
+            for (int i = 0; i < floatArr.Length; i++)
+            {
+                bool success = float.TryParse(balances[i], out float number);
+                if (success) floatArr[i - failLength] = number;
+                else failLength++;
+            }
+            if(failLength == 0) return floatArr;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 13]
```diff
@@ -1,37 +1,261 @@
-﻿namespace BootCamp.Chapter
+﻿using System.Text;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Analyzer 'SonarAnalyzer.Rules.CSharp.ObjectsShouldNotBeDisposedMoreThanOnce' threw an exception of type 'SonarAnalyzer.SymbolicExecution.SymbolicExecutionException' with message 'Error processing method: FindRichestPerson ## Method file: BalanceStats.cs ## Method line: 56,8 ## Inner exception: System.InvalidOperationException: This operation does not apply to an empty instance. ##   at System.Collections.Immutable.ImmutableStack`1[T].Peek () [0x0000d] in <2e5af6ef122c4468962237668e6f8259>:0  ##   at System.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 14]
```diff
@@ -1,37 +1,261 @@
...
+        /// </summary>
+        public static float[] StringArrToFloatArr(string[] balances)
+        {
+            int failLength = 0;
+            float[] floatArr = new float[balances.Length];
+            for (int i = 0; i < floatArr.Length; i++)
+            {
+                bool success = float.TryParse(balances[i], out float number);
+                if (success) floatArr[i - failLength] = number;
+                else failLength++;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'else' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 15]
```diff
@@ -1,37 +1,261 @@
...
+                return message;
+            }
+            for (int i = 1; i < workingBalances.Length; i++)
+            {
+                if(float.Parse(workingBalances[i][^1]) > float.Parse(workingBalances[richestPerson][^1]))
+                {
+                    richestPerson = i;
+                }
+            }
+            return message = $"{workingBalances[richestPerson][0]} is the richest person. ${workingBalances[richestPerson][^1]}";
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Remove this useless assignment to local variable 'message'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs (hunk 1 above)
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

**Comment by TelefunkenU87** [comment]
Ok, I see a lot of problems to address. I will start working on them soon.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/BalanceStats.cs  4
- Src/BootCamp.Chapter/TextTable.cs  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603&bid=16933732)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 16]
```diff
@@ -1,37 +1,288 @@
...
+        {
+            int count = balanceNames.Split(',').Length - 1;
+            balanceNames = balanceNames.Remove(balanceNames.Length - 2);
+            if (count == 1)
+            {
+                return balanceNames;
+            }
+            string[] names = balanceNames.Split(',');
+            StringBuilder sbNames = new StringBuilder(names[0]);
+            switch (count)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Replace this 'switch' statement with 'if' statements to increase readability.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 17]
```diff
@@ -1,37 +1,288 @@
...
+        {
+            string[][] workingPeopleAndBalances = ArrayifyBalances(peopleAndBalances);
+            string totalNames = "";
+            for (int i = 0; i < workingPeopleAndBalances.Length; i++)
+            {
+                float[] balances = StringArrToFloatArr(workingPeopleAndBalances[i][1..]);
+                bool isBalance = false;
+                foreach (var item in balances)
+                {
+                    if (item == balance)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Do not check floating point equality with exact values, use a range instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/TextTable.cs [hunk 18]
```diff
@@ -34,7 +36,51 @@ public static class TextTable
...
```
Comment (Almantask) [inline Src/BootCamp.Chapter/TextTable.cs]: Codacy Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 19]
```diff
@@ -1,37 +1,288 @@
...
+                    {
+                        isBalance = true;
+                    }
+                }
+                if (isBalance)
+                {
+                    totalNames += $"{workingPeopleAndBalances[i][0]}, ";
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Use a StringBuilder instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 20]
```diff
@@ -1,37 +1,288 @@
-﻿namespace BootCamp.Chapter
+﻿using System.Text;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Analyzer 'SonarAnalyzer.Rules.CSharp.ObjectsShouldNotBeDisposedMoreThanOnce' threw an exception of type 'SonarAnalyzer.SymbolicExecution.SymbolicExecutionException' with message 'Error processing method: FindRichestPerson ## Method file: BalanceStats.cs ## Method line: 56,8 ## Inner exception: System.InvalidOperationException: This operation does not apply to an empty instance. ##   at System.Collections.Immutable.ImmutableStack`1[T].Peek () [0x0000d] in <2e5af6ef122c4468962237668e6f8259>:0  ##   at System.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/BalanceStats.cs  4
- Src/BootCamp.Chapter/TextTable.cs  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603&bid=16933732)

File: Src/BootCamp.Chapter/TextTable.cs (hunk 18 above)
Comment (Almantask) [inline Src/BootCamp.Chapter/TextTable.cs]: Codacy Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 21]
```diff
@@ -1,37 +1,288 @@
...
+                foreach (var item in balances)
+                {
+                    if (item == balance)
+                    {
+                        isBalance = true;
+                    }
+                }
+                if (isBalance)
+                {
+                    totalNames += $"{workingPeopleAndBalances[i][0]}, ";
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Use a StringBuilder instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs (hunk 20 above)
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Analyzer 'SonarAnalyzer.Rules.CSharp.ObjectsShouldNotBeDisposedMoreThanOnce' threw an exception of type 'SonarAnalyzer.SymbolicExecution.SymbolicExecutionException' with message 'Error processing method: FindRichestPerson ## Method file: BalanceStats.cs ## Method line: 56,8 ## Inner exception: System.InvalidOperationException: This operation does not apply to an empty instance. ##   at System.Collections.Immutable.ImmutableStack`1[T].Peek () [0x0000d] in <2e5af6ef122c4468962237668e6f8259>:0  ##   at System.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 22]
```diff
@@ -1,37 +1,288 @@
...
+        {
+            string[][] workingPeopleAndBalances = ArrayifyBalances(peopleAndBalances);
+            string totalNames = "";
+            for (int i = 0; i < workingPeopleAndBalances.Length; i++)
+            {
+                float[] balances = StringArrToFloatArr(workingPeopleAndBalances[i][1..]);
+                bool isBalance = false;
+                foreach (var item in balances)
+                {
+                    if (item == balance)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Do not check floating point equality with exact values, use a range instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 23]
```diff
@@ -1,37 +1,288 @@
...
+        {
+            int count = balanceNames.Split(',').Length - 1;
+            balanceNames = balanceNames.Remove(balanceNames.Length - 2);
+            if (count == 1)
+            {
+                return balanceNames;
+            }
+            string[] names = balanceNames.Split(',');
+            StringBuilder sbNames = new StringBuilder(names[0]);
+            switch (count)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Replace this 'switch' statement with 'if' statements to increase readability.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/BalanceStats.cs  5
- Src/BootCamp.Chapter/TextTable.cs  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603&bid=16933732)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 24]
```diff
@@ -1,37 +1,291 @@
-﻿namespace BootCamp.Chapter
+﻿using System.Text;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Analyzer 'SonarAnalyzer.Rules.CSharp.ObjectsShouldNotBeDisposedMoreThanOnce' threw an exception of type 'SonarAnalyzer.SymbolicExecution.SymbolicExecutionException' with message 'Error processing method: FindRichestPerson ## Method file: BalanceStats.cs ## Method line: 56,8 ## Inner exception: System.InvalidOperationException: This operation does not apply to an empty instance. ##   at System.Collections.Immutable.ImmutableStack`1[T].Peek () [0x0000d] in <2e5af6ef122c4468962237668e6f8259>:0  ##   at System.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 25]
```diff
@@ -1,37 +1,291 @@
...
         {
-            return "";
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return invalidMessage;
+            }
+            string[][] workingPeopleAndBalances = ArrayifyBalances(peopleAndBalances);
+            for (int i = 0; i < workingPeopleAndBalances.Length; i++)
+            {
+                if (workingPeopleAndBalances[i].Length <= 2)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 26]
```diff
@@ -1,37 +1,291 @@
...
+        {
+            int count = balanceNames.Split(',').Length - 1;
+            balanceNames = balanceNames.Remove(balanceNames.Length - 2);
+            if (count == 1)
+            {
+                return balanceNames;
+            }
+            string[] names = balanceNames.Split(',');
+            StringBuilder sbNames = new StringBuilder(names[0]);
+            switch (count)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Replace this 'switch' statement with 'if' statements to increase readability.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 27]
```diff
@@ -1,37 +1,291 @@
...
+        {
+            string[][] workingPeopleAndBalances = ArrayifyBalances(peopleAndBalances);
+            string totalNames = "";
+            for (int i = 0; i < workingPeopleAndBalances.Length; i++)
+            {
+                float[] balances = StringArrToFloatArr(workingPeopleAndBalances[i][1..]);
+                bool isBalance = false;
+                foreach (var item in balances)
+                {
+                    if (item == balance)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Do not check floating point equality with exact values, use a range instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 28]
```diff
@@ -1,37 +1,291 @@
...
+                    {
+                        isBalance = true;
+                    }
+                }
+                if (isBalance)
+                {
+                    totalNames += $"{workingPeopleAndBalances[i][0]}, ";
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Use a StringBuilder instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/TextTable.cs (hunk 18 above)
Comment (Almantask) [inline Src/BootCamp.Chapter/TextTable.cs]: Codacy Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 29]
```diff
@@ -1,37 +1,291 @@
...
         /// </summary>
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return invalidMessage;
+            }
+
+            float highestBalanceEver = HighestBalanceEver(peopleAndBalances);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Method names should be verbs, not nouns.

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 30]
```diff
@@ -1,37 +1,291 @@
...
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return invalidMessage;
+            }
+            string[][] workingPeopleAndBalances = ArrayifyBalances(peopleAndBalances);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Please pick better name

File: Src/BootCamp.Chapter/TextTable.cs [hunk 31]
```diff
@@ -34,7 +36,51 @@ public static class TextTable
         /// </summary>
         public static string Build(string message, int padding)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/TextTable.cs]: Try to split this method into parts. For example into parts where you add top, bottom, left and right frames, where you determine the count of dashes, where you populate the frame with a message.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/BalanceStats.cs  5
- Src/BootCamp.Chapter/TextTable.cs  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603&bid=16933732)

File: Src/BootCamp.Chapter/TextTable.cs (hunk 18 above)
Comment (Almantask) [inline Src/BootCamp.Chapter/TextTable.cs]: Codacy Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 32]
```diff
@@ -1,37 +1,329 @@
...
+                {
+                    loss = balance[i + 1] - balance[i];
+                }
+            }
+            return loss;
+        }
+
+        public static string ConcatMoney(float balance)
+        {
+            if (balance < 0)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Use the '?:' operator here.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 33]
```diff
@@ -1,37 +1,329 @@
...
+                float[] balances = ConvertStringArrToFloatArr(workingPeopleAndBalances[i][1..]);
+                bool isBalance = false;
+                int isCurrentIndexShift = 0;
+                if (isCurrent)
+                {
+                    isCurrentIndexShift = balances.Length - 1;
+                }
+                for (int j = 0 + isCurrentIndexShift; j < balances.Length; j++)
+                {
+                    if(balances[j] == balance)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Do not check floating point equality with exact values, use a range instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 34]
```diff
@@ -1,37 +1,329 @@
...
+                    {
+                        isBalance = true;
+                    }
+                }
+                if (isBalance)
+                {
+                    totalNames += $"{workingPeopleAndBalances[i][0]}, ";
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Use a StringBuilder instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/BalanceStats.cs  5
- Src/BootCamp.Chapter/TextTable.cs  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603&bid=16933732)

File: Src/BootCamp.Chapter/BalanceStats.cs (hunk 33 above)
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Do not check floating point equality with exact values, use a range instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs (hunk 32 above)
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Use the '?:' operator here.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs (hunk 34 above)
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Use a StringBuilder instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/TextTable.cs (hunk 18 above)
Comment (Almantask) [inline Src/BootCamp.Chapter/TextTable.cs]: Codacy Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/BalanceStats.cs  5
- Src/BootCamp.Chapter/TextTable.cs  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603&bid=16933732)

File: Src/BootCamp.Chapter/TextTable.cs (hunk 18 above)
Comment (Almantask) [inline Src/BootCamp.Chapter/TextTable.cs]: Codacy Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 35]
```diff
@@ -1,37 +1,329 @@
...
+                {
+                    loss = balance[i + 1] - balance[i];
+                }
+            }
+            return loss;
+        }
+
+        public static string ConcatMoney(float balance)
+        {
+            if (balance < 0)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Use the '?:' operator here.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs (hunk 33 above)
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Do not check floating point equality with exact values, use a range instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

File: Src/BootCamp.Chapter/BalanceStats.cs (hunk 34 above)
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Use a StringBuilder instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)

**Comment by TelefunkenU87** [comment]
Ok, I got all the tests wrapped up. I'll loop back arround and address the changes you requested and see what I can do about some of these Codacy concerns.

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 36]
```diff
@@ -1,37 +1,329 @@
...
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return invalidMessage;
+            }
+            string[][] workingPeopleAndBalances = ReturnJaggedArray(peopleAndBalances);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: So what that you return a jagged array? Is it hardcoded? Is it made of something? PLease rename better,. For  example: ConvertToJaggedArray is a way better name.

**Pull request closed without merging**